src/commands/new.ts

```diff
@@ -1,130 +1,142 @@
+import { repoUrl } from '#constants';
 import { CommandExists, CreateFileFromTemplate } from '#functions';
 import { PromptNew } from '#prompts';
-import chalk from 'chalk';
-import { exec, spawn } from 'child_process';
-import { existsSync } from 'fs';
-import { cp, readFile, rm, writeFile } from 'fs/promises';
+import { fromAsync, isErr } from '@sapphire/result';
+import { blueBright, red } from 'colorette';
+import { execa } from 'execa';
+import { existsSync } from 'node:fs';
+import { cp, readFile, rm, writeFile } from 'node:fs/promises';
+import { resolve } from 'node:path';
 import ora from 'ora';
-import { resolve } from 'path';
 import prompts from 'prompts';
-import { repoUrl } from '#constants';
 
-const { blueBright, red } = chalk;
+async function editPackageJson(location: string, name: string) {
+	const pjLocation = `./${location}/package.json`;
+	const output = JSON.parse(await readFile(pjLocation, 'utf8'));
+	if (!output) throw new Error("Can't read file.");
+
+	output.name = name;
 
-function editPackageJson(location: string, name: string) {
-	return new Promise(async (resolve, reject) => {
-		const pjLocation = `./${location}/package.json`;
-		const output = JSON.parse(await readFile(pjLocation, 'utf8'));
-		if (!output) return reject(new Error("Can't read file."));
+	const result = await fromAsync(async () => writeFile(pjLocation, JSON.stringify(output, null, 2)));
+
+	if (isErr(result)) {
+		return false;
+	}
 
-		output.name = name;
-		await writeFile(pjLocation, JSON.stringify(output, null, 2)).catch(reject);
-		return resolve(true);
-	});
+	return true;
 }
 
-function installDeps(location: string, pm: string, verbose: boolean) {
-	const pmp = spawn(process.platform === 'win32' ? `${pm.toLowerCase()}.cmd` : pm.toLowerCase(), ['install'], {
-		stdio: verbose ? 'inherit' : undefined,
-		cwd: `./${location}/`
-	});
-
-	return new Promise((resolve, reject) => {
-		pmp.on('error', reject);
-
-		pmp.on('exit', async (code) => {
-			if (code === 0) {
-				const lockfile = `./${location}/${pm === 'npm' ? 'yarn.lock' : 'package-lock.json'}`;
-				if (existsSync(lockfile)) await rm(lockfile);
-				resolve(true);
-			} else {
-				reject(new Error('An unknown error occured while installing the dependencies. Try running Sapphire CLI with "--verbose" flag.'));
-			}
-		});
-	});
+async function installDeps(location: string, pm: string, verbose: boolean) {
+	const result = await fromAsync(async () =>
+		execa(process.platform === 'win32' ? `${pm.toLowerCase()}.cmd` : pm.toLowerCase(), ['install'], {
```

Comment:
Why are you doing .cmd again? Between volta, corepack and w/e, nobody uses .cmd for package manages